Assets/Scripts/UI/PickupPup/ShelterScreen/PPShelterUIController.cs

```diff
@@ -17,14 +20,31 @@ protected override void setReferences()
         promptID = PromptID.ShelterPrompt;
         base.setReferences();
         availableDogPortraits = GetComponentsInChildren<DogSlot>();
+		if (dogProfileObject != null)
+        {
+			dogProfileObject.SetActive(false);
+        }
+		if (!dogShelterProfile)
+		{
+			dogShelterProfile = dogProfileObject.GetComponent<DogShelterProfile>();
+		}
+		if (!dogProfileButtonController)
+		{
+			dogProfileButtonController = dogProfileObject.GetComponent<DogProfileButtonController>();
+		}
     }
 
     protected override void fetchReferences()
     {
         base.fetchReferences();
         database = DogDatabase.GetInstance;
         EventController.Event(PPEvent.LoadShelter);
-        populateAvailableDogs(database);
+        var dogs = populateAvailableDogs(database);
```

Comment:
Think we should avoid use of `var` when possible. Prefer explicitly indicating type